deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/impl/V1PodHandler.java

```diff
@@ -372,4 +393,122 @@ public boolean cleanUp() {
 	public int getWeight() {
 	    return ResourceKind.POD.getWeight();
 	}
+	
+	public void watch(ApiClient apiClient, String appName, String serviceName, String namespace)
+			throws IOException, ApiException, HyscaleException {
+		OkHttpClient httpClient = apiClient.getHttpClient().newBuilder().readTimeout(0, TimeUnit.SECONDS).build();
+		apiClient.setHttpClient(httpClient);
+		CoreV1Api api = new CoreV1Api(apiClient);
+
+		String selector = ResourceSelectorUtil.getServiceSelector(appName, serviceName);
+		
+		Integer replicas = null;
+		String latestPodSelector = null;
+		V1StatefulSet v1StatefulSet=null;
+		try {
+			v1StatefulSet=getV1StatefulSetHandler().get(apiClient, serviceName, namespace);
+		}catch(Exception e) {
+			LOGGER.info("service is not statefulset");
+		}
+		if (v1StatefulSet!=null) {
+			latestPodSelector=getV1StatefulSetHandler().getPodSelector(v1StatefulSet.getStatus(), selector);
+			replicas = getV1StatefulSetReplicas(v1StatefulSet);
```

Comment:
Since its a single line function we don't need method.

Reply:
I hope you go my previous comments , there should be a separate method to get the parent as String. From this string resolve the pod parent handler from the PodParentFactory ..with this parent handler call the podSelector and then merge.